src/app/hierarchical-grid/hierarchical-grid-validator-service-extended/hierarchical-grid-validator-service-extended.component.html

```diff
@@ -0,0 +1,75 @@
+            <igx-column [editable]="false" field="OrderID"></igx-column>
+            <igx-column [editable]="true" field="EmployeeID" [editable]="true" required></igx-column>
+            <igx-column [editable]="true" field="OrderDate" [dataType]="'date'" [editable]="true" required>
+                <ng-template igxCellValidationError let-cell='cell'>
+                    <ng-container *ngTemplateOutlet="cell.defaultErrorTemplate">
+                    </ng-container>
+                    <div *ngIf="cell.errors?.['beyondThreshold']">
+                        The date cannot be in the future.
+                    </div>
+                </ng-template>
+            </igx-column>
+            <igx-column [editable]="true" field="RequiredDate" [dataType]="'date'" [editable]="true" required>
+                <ng-template igxCellValidationError let-cell='cell'>
+                    <ng-container *ngTemplateOutlet="cell.defaultErrorTemplate">
+                    </ng-container>
+                    <div *ngIf="cell.errors?.['futureDate']">
+                        The date cannot be in the future.
+                    </div>
+                    <div *ngIf="cell.errors?.['pastDate']">
+                        The date cannot be before the 5th of November 2010
+                    </div>
+                </ng-template>
+            </igx-column>
+            <igx-column field="ShippedDate" [dataType]="'date'" [editable]="true">
+                <ng-template igxCellValidationError let-cell='cell'>
+                    <ng-container *ngTemplateOutlet="cell.defaultErrorTemplate">
+                    </ng-container>
+                    <div *ngIf="cell.errors?.['futureDate']">
+                        The date cannot be in the future.
+                    </div>
+                    <div *ngIf="cell.errors?.['pastDate']">
+                        The date cannot be before the 5th of November 2010
+                    </div>
+                </ng-template>
+            </igx-column>
+            <igx-column field="ShipVia" [selectable]="false" [editable]="true"></igx-column>
+            <igx-column field="Freight" [selectable]="false" [editable]="true"></igx-column>
+
+            <igx-row-island [height]="null" [key]="'OrderDetails'" [primaryKey]="'ProductID'" [autoGenerate]="false"
+                [batchEditing]="true" [rowEditable]="true" [primaryKey]="'ProductID'">
+                <igx-column [editable]="true" field="ProductID" [editable]="false" required></igx-column>
+                <igx-column [editable]="true" field="UnitPrice" [editable]="true" required></igx-column>
+                <igx-column [editable]="true" field="Quantity" [editable]="true" required></igx-column>
+                <igx-column [editable]="true" field="Discount" [editable]="true"></igx-column>
+            </igx-row-island>
+        </igx-row-island>
+    </igx-hierarchical-grid>
+
+    <div class="buttons-wrapper">
```

Comment:
It would be nice to have buttons for the child grids as well, maybe in the toolbar similar to this sample:
https://www.infragistics.com/products/ignite-ui-angular/angular/components/hierarchicalgrid/batch-editing#angular-hierarchical-grid-batch-editing-and-transactions-example